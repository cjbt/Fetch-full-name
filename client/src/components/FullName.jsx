import React, { useState, useEffect } from 'react';
import axios from 'axios';

const FullName = () => {
  const [firstname, setFirstname] = useState('');
  const [middlename, setMiddlename] = useState('');
  const [lastname, setLastname] = useState('');

  useEffect(() => {
    fetchFullName();
  }, []);

  const fetchFullName = async () => {
    try {
      const first = await axios.get('http://localhost:8000/firstname');
      const middle = await axios.get('http://localhost:8000/middlename');
      const last = await axios.get('http://localhost:8000/lastname');
      setFirstname(first.data.firstname);
      setMiddlename(middle.data.middlename);
      setLastname(last.data.lastname);
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <div>
      <h1>
        {firstname} {middlename} {lastname}
      </h1>
    </div>
  );
};

export default FullName;
