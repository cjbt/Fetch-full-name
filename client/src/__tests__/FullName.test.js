import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FullName from '../components/FullName';

afterEach(cleanup);

it('matches snapshot test', () => {
  const { asFragment } = render(<FullName />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders Full name after component mounts to the dom', async () => {
  const { findByText, queryByText } = render(<FullName />);

  const beforeUseEffect = queryByText(/Jane Smith Doe/i);
  expect(beforeUseEffect).not.toBeInTheDocument();

  const afterUseEffect = findByText(/Jane Smith Doe/i);
  expect(await afterUseEffect).toBeInTheDocument();
});
