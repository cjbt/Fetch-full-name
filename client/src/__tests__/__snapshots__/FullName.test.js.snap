// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot test 1`] = `
<DocumentFragment>
  <div>
    <h1>
        
    </h1>
  </div>
</DocumentFragment>
`;
